cogs/moderation.py

```diff
@@ -1,28 +1,53 @@
 import asyncio
 import textwrap
+import typing as t
+from collections import Counter
 
 import discord
 from discord import Color, Embed
 from discord.ext.commands import (
+    BadArgument,
     Bot,
     Cog,
     Context,
+    Converter,
+    Greedy,
+    MemberConverter,
     bot_has_permissions,
     command,
+    guild_only,
     has_permissions,
 )
 
+from .utils.formats import Plural
 
+
+def can_execute_action(ctx: Context, user: discord.User, target: discord.User) -> bool:
+    return user.id == ctx.bot.owner_id or user == ctx.guild.owner or user.top_role > target.top_role
+
+
+class MemberNotFound(Exception):
+    pass
+
+
+async def resolve_user(bot: Bot, member_id: int) -> discord.User:
+    user = bot.get_member(member_id)
+    if user is None:
+        raise MemberNotFound()
+    return user
+
+
+# TODO : Clean everything up
 class Moderation(Cog):
     """This cog provides moderation commands."""
 
-    def __init__(self, bot: Bot) -> None:
+    def __init__(self, bot: discord.Bot) -> None:
```

Comment:
Why did you do this? what's wrong with `bot: Bot` there is no point in using `discord.Bot` it's the same but longer.